packages/remix-dev/config/routesConvention.ts

```diff
@@ -21,20 +22,24 @@ export function isRouteModuleFile(filename: string): boolean {
  * For example, a file named `app/routes/gists/$username.tsx` creates a route
  * with a path of `gists/:username`.
  */
-export function defineConventionalRoutes(appDir: string): RouteManifest {
+export function defineConventionalRoutes(
+  appDir: string,
+  ignoredFilePatterns?: string[]
+): RouteManifest {
   let files: { [routeId: string]: string } = {};
 
   // First, find all route modules in app/routes
   visitFiles(path.join(appDir, "routes"), file => {
-    let routeId = createRouteId(path.join("routes", file));
-
-    if (isRouteModuleFile(file)) {
-      files[routeId] = path.join("routes", file);
+    if (
+      ignoredFilePatterns &&
+      ignoredFilePatterns.some(pattern => minimatch(file, pattern))
```

Comment:
No need for a check if you give `ignoredFilePatterns` a default value of `[]`

```suggestion
      ignoredFilePatterns.some(pattern => minimatch(file, pattern))
```